qiskit_ibm_runtime/options/options.py

```diff
@@ -106,15 +115,20 @@ class Options:
         "execution": ExecutionOptions,
         "environment": EnvironmentOptions,
         "simulator": SimulatorOptions,
+        "resilience": ResilienceOptions,
     }
 
     @staticmethod
-    def _get_program_inputs(options: dict) -> dict:
+    def _get_program_inputs(
+        options: dict, primitive: PrimitiveType, backend: IBMBackend = None
+    ) -> dict:
         """Convert the input options to program compatible inputs.
 
         Returns:
             Inputs acceptable by primitive programs.
         """
+        if not os.getenv("QISKIT_RUNTIME_REMOVE_OPTIONS_VALIDATION"):
+            Options._validate_program_inputs(options, primitive, backend)
```

Comment:
I'd recommend calling `_validate_program_inputs` separately from `_get_program_inputs` as to not complicate the functionality of `_get_program_inputs` which just converts options to program inputs.